optaplanner-core/src/main/java/org/optaplanner/core/impl/domain/solution/util/Cache.java

```diff
@@ -0,0 +1,30 @@
+package org.optaplanner.core.impl.domain.solution.util;
+
+import java.util.function.Function;
+
+/**
+ * Simple cache interface.
+ *
+ * @param <K> the type of keys for the cache
+ * @param <V> the type of values stored in the cache
+ * */
+public interface Cache<K, V> {
```

Comment:
Did you consider extending `ConcurrentHashMap`, overriding `computeIfAbsent(...)`? Seems that introducing this interface is not strictly necessary.

Reply:
I did - I didn't like changing semantic of JDK method a bit, but maybe it's not so bad thing. Thanks for suggestion, I will change it.